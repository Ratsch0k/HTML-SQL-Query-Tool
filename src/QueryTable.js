import React from 'react';
import Table from 'react-bootstrap/Table'

function QueryTable (props) {
    if(props.data === null){
        return null;
    }else {
        // Get keys of data
        const keys = Object.keys(props.data[0]);
        let rows = [];

        // Generate head of table
        let cells = [];
        keys.forEach((name) => cells.push(<th key={name}>{name}</th>));
        const head = <tr>{cells}</tr>;

        // Fill body of table
        let r = 0, i = 0;
        props.data.forEach((rowData) => {
            cells = [];
            // Fill row
            Object.values(rowData).forEach((value) => {
               cells.push(<td key={`row-${r} cell-${i}`}>{value}</td>);
               i++;
            });
            rows.push(<tr key={`row-${r}`}>{cells}</tr>);
            i = 0;
            r++;
        });

        return (
            <Table striped hover borderless size='sm'>
                <thead>
                    {head}
                </thead>
                <tbody>
                    {rows}
                </tbody>
            </Table>
        )
    }
}

export default QueryTable;