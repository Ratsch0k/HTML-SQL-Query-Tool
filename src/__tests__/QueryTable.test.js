import React from 'react';
import ReactDom from 'react-dom';
import QueryTable from '../QueryTable';
import renderer from 'react-test-renderer';

const testData1 = [
    {name: "Simon", age: 20},
    {name: "Annika", age: 20},
    {name: "Lisa", age: 18},
];

const testData2 = [
    {email: "test@web.de", username: "rat", age: 43, password: "password"},
    {email: "user@yahoo.it", username: "user", age: 17, password: "asdfad"},
    {email: "tree@o2.de", username: "tree", age: 18, password: "asdfasdfasw"},
    {email: "benutzer@gmail.com", username: "benutzer", age: 98, password: "afsthtr"},
];


it('QueryTable with data renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<QueryTable data={testData1}/>, div);
    ReactDom.unmountComponentAtNode(div);
});

it('QueryTable without data renders without crashes', () => {
    const div = document.createElement('div');
    ReactDom.render(<QueryTable/>, div);
    ReactDom.unmountComponentAtNode(div);
})

it('QueryTable changes renders new if data changes', () => {
    let component = renderer.create(<QueryTable/>);
    let tree;

    // First render without data
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Second render with first test data
    component = renderer.create(<QueryTable data={testData1}/>);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Third render with second test data
    tree.props.data = testData2;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});