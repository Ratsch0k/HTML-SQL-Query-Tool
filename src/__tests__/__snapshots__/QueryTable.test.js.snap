// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryTable changes renders new if data changes 1`] = `null`;

exports[`QueryTable changes renders new if data changes 2`] = `
<div
  className="MuiBox-root MuiBox-root-53"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-81"
    >
      <table
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              NAME
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              AGE
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Simon
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              20
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Annika
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              20
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Lisa
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              18
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`QueryTable changes renders new if data changes 3`] = `
<div
  className="MuiBox-root MuiBox-root-53"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-81"
    >
      <table
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              NAME
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              AGE
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Simon
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              20
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Annika
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              20
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Lisa
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              18
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
